sled-agent/src/sled_agent.rs

```diff
@@ -554,6 +552,12 @@ impl SledAgent {
         Ok(())
     }
 
+    /// Gets the sled's current list of all zpools.
+    pub async fn zpools_get(&self) -> Result<Vec<Zpool>, Error> {
+        let zpools = self.inner.storage.get_zpools().await?;
+        Ok(zpools)
+    }
```

Comment:
This is a new endpoint being exposed from the sled agent so RSS can query for zpool information when deciding "where should CRDB go?"